master/internal/api_user_intg_test.go

```diff
@@ -924,3 +926,207 @@ func getActivityEntry(ctx context.Context, userID model.UserID, entityID int32)
 	return db.Bun().NewSelect().Model((*model.UserActivity)(nil)).Where("user_id = ?",
 		int32(userID)).Where("entity_id = ?", entityID).Count(ctx)
 }
+
+func TestPostLongLivedToken(t *testing.T) {
+	api, curUser, ctx := setupAPITest(t, nil)
+
+	// Without lifespan input
+	token, err := api.PostLongLivedToken(ctx, &apiv1.PostLongLivedTokenRequest{})
+	require.NoError(t, err)
+	require.NotNil(t, token)
+
+	exists, err := getLongLivedTokensEntry(ctx, curUser.ID)
+	require.True(t, exists)
+	require.NoError(t, err)
+}
+
+func TestPostLongLivedTokenWithLifespan(t *testing.T) {
```

Comment:
nit (non-blocking): We could consolidate these tests using table-based structured testing where we have individual test cases represented in an array/map of structs that contain the necessary information; maybe something like:
```
type testCases struct {
   description string 
   lifespan *string 
   expected LongLivedTokenInfo
   expectedError *error
}
```

Reply:
Absolutely fair, that's part of the reason why I'd encourage the use of the description to capture that information & making adding additional test cases as simple as adding a new entry into the struct. 

However, it could be worth just adding a helper function to handle the validation of the result since it does look like there's quite a bit of repetition when it comes to checking the Token entries.

Reply:
Noted! Will make it more modular.

Reply:
I tried making it in a tabular structure, but because each test had different parameters and APIs, it ended up being more cluttered and massive. So, I chose to make it more modular and provide comments.